doc/arch/adr-002-decide-how-to-display-more-facilities.md

```diff
@@ -0,0 +1,233 @@
+would add a `/tile/{layer}/{z}/{x}/{y}/` endpoint to the Django application,
+then update the client to make tile requests there rather than rendering the
+`/facilities` GeoJSON response as Leaflet markers.
+
+#### Pros
+
+##### Provides Access to the Existing Django Queryset Apparatus
+
+While using Martin (or a similar solution) would compel writing new versions of
+the facilities queries in PL/pgSQL, placing a vector tile endpoint in Django
+lets us reuse some of the existing query code and also provides access to
+Django models and querysets. Likewise, we would not have to write new code for
+new filter and search options in two different languages.
+
+##### Already Has a Secure Database Connection
+
+The Django web application already has a secure database connection, so we
+would not have to create a solution for securing Martin or another PostGIS-backed
+tile server.
+
+##### Enables Switching from `ST_AsMVT` to Another Python Vector Tile Option
+
+There remains some question about the viability of using `ST_AsMVT`. If it turns
+out that this is not a performant solution, having the tile endpoint in Django
+makes it possible to drop out of using `ST_AsMVT` altogether and instead using
+an alternate Python library for generating the vector tiles.
+
+##### Doesn't Require Creating & Deploying a Different Service
+
+Adding Martin or another vector tile server would increase the number of
+different kinds of services running as part of the OAR, which adds to the
+application's complexity. Keeping the tile endpoint in Django does not require
+adding a new service.
+
+##### Allows Scaling by Increasing the Number of App Instances
+
+Adding a tile endpoint to the Django app also enables continuing to scale the
+application in the usual way: by increasing the number of app instances
+available to serve requests.
+
+#### Cons
+
+##### Mingles Tile Request Traffic with Other App Traffic
+
+The biggest downside of adding a vector tile endpoint to the Django app is
+that it would mean mingling tile request traffic with other app traffic.
+While we plan to limit the size of tile request responses by clusting facilities
+at different zoom levels, tile request traffic will likely be more frequent and
+sustained than requests to the current `/facilities` endpoint.
+
+## Decision
+
+We have decided to add a vector tile endpoint to the existing Django app.
+
+While Martin, in particular, seemed like a compelling solution, we had enough
+questions about using it to discourage us from taking on the complexity of
+using it here.
+
+Our main apprehension about adding a tile endpoint to the existing web app is
+that it'll mingle tile requests with other requests in a way that could cause
+performance problems. However, given the size of the OAR's traffic and the
+possibility of addressing traffic increases by scaling the number of app
+instances, this seemed like an acceptable tradeoff.
+
+## Consequences
+
+As a consequence of this decision, we will need to:
```

Comment:
Potentially too low detail for the ADR, but we may also need to consider:

- Tweaking the number of Gunicorn workers per Gunicorn instance, in addition to the Gunicorn worker type (synchronous vs. Gevent). This would be in service of trying to increase the concurrent request capabilities of a single Gunicorn instance vs. spinning up more of them.
- Looking at caching HTTP headers used by Whitenoise for static assets and evaluating which make sense to replicate for `/tile`.
- Extending [caching behaviors](https://github.com/open-apparel-registry/open-apparel-registry/blob/develop/deployment/terraform/cdn.tf#L43-L62) in CloudFront so that they respect any caching header changes made at the Django level.

Reply:
Sounds good -- I'll add line items about Gunicorn configuration & caching to the list.